zetaclient/chains/evm/observer/v2_inbound.go

```diff
@@ -175,6 +185,12 @@ func (ob *Observer) newDepositInboundVote(event *gatewayevm.GatewayEVMDeposited)
 		isCrossChainCall = true
 	}
 
+	// determine confirmation mode
+	confirmationMode := types.ConfirmationMode_FAST
+	if ob.IsBlockConfirmedForInboundSafe(event.Raw.BlockNumber) {
+		confirmationMode = types.ConfirmationMode_SAFE
+	}
```

Comment:
This is when IsInboundEligibleForFastConfirmation should be used

It is unclear to me why we use IsBlockConfirmedForInboundSafe

Reply:
The same situation for caller. The caller (holding the vote message) needs to decide skip the vote or not.